src/rust/engine/src/core.rs

```diff
@@ -88,6 +88,10 @@ impl hash::Hash for Key {
 }
 
 impl Key {
+  pub fn new_with_anon_type_id(id: Id) -> Key {
+    Key { id: id, type_id: TypeId(0) }
```

Comment:
it's int on the python side and named `anon_id` as https://github.com/pantsbuild/pants/blob/master/src/python/pants/engine/subsystem/native.py#L390-L391